.github/workflows/build-server.yaml

```
@@ -130,56 +130,92 @@ jobs:
-        run: |
-          # Push image to AWS ECR
-          aws ecr get-login-password --region $AWS_DEFAULT_REGION | docker login --username AWS --password-stdin $AWS_ACCOUNT_ID.dkr.ecr.$AWS_DEFAULT_REGION.amazonaws.com
-          docker tag $CONTAINER_IMAGE-dev:${{ github.sha }} $AWS_ACCOUNT_ID.dkr.ecr.$AWS_DEFAULT_REGION.amazonaws.com/$CONTAINER_IMAGE-dev:${{ github.sha }}
+          # Push both main image and cache
           docker push $AWS_ACCOUNT_ID.dkr.ecr.$AWS_DEFAULT_REGION.amazonaws.com/$CONTAINER_IMAGE-dev:${{ github.sha }}
+          docker push $CACHE_IMAGE
 
-      - name: Push (DEMO)
+      - name: Build and push image (DEMO) - With Registry Caching
         if: github.ref == 'refs/heads/demo'
         run: |
```

actionlint (1.7.7)

shellcheck reported issue in this script: SC2086:info:2:37: Double quote to prevent globbing and word splitting

(shellcheck)